### PR TITLE
Adds a unit test to detect double stacked lights

### PR DESCRIPTION
**Pull request by LemonInTheDark**
## About The Pull Request

They make me sad and we should test for them
I use dir here to prevent like, bulb + bar stuff, idk if that's wanted or not tho

### CODE REVIEW

**Review by Mothblocks — APPROVED** [review]
You can probably do this with maplint too (or at least should be able to), but I have no preference and this is more stable

File: code/modules/power/lighting/light.dm [hunk 1]
```diff
@@ -88,6 +88,17 @@
...
+	// Detect and scream about double stacked lights
+	if(PERFORM_ALL_TESTS(focus_only/stacked_lights))
+		var/turf/our_location = get_turf(src)
+		for(var/obj/machinery/light/on_turf in our_location)
+			if(on_turf == src)
+				continue
+			if(on_turf.dir != dir)
+				continue
+			stack_trace("Conflicting double stacked light [on_turf.type] found at ([our_location.x],[our_location.y],[our_location.z])")
+			qdel(on_turf)
```
Comment (JohnFulpWillard) [inline code/modules/power/lighting/light.dm:100]: Why're we deleting the turf?

Comment (san7890) [reply code/modules/power/lighting/light.dm:100]: image

whar

Comment (JohnFulpWillard) [reply code/modules/power/lighting/light.dm:100]: oh lol i misread

Comment (LemonInTheDark) [reply code/modules/power/lighting/light.dm:100]: means it might be a bad name

**Pull request merged**